core/templates/dev/head/pages/admin/roles_tab/role_graph_directive.html

```diff
@@ -0,0 +1,70 @@
+<!-- NOTE TO DEVELOPERS: The svg defs needed for this directive are
+included at the bottom of exploration_editor.html. In order for the
+elements to render properly in Firefox, there cannot be multiple defs
+with the same id on the page. -->
+
+<style>
+  .oppia-graph-viz path.link {
+    fill: none;
+    stroke: rgb(179, 179, 179);
+    stroke-width: 3px;
+  }
+  .oppia-graph-viz text {
+    font: 12px sans-serif;
+    pointer-events: none;
+  }
+  .oppia-graph-viz text.shadow {
+    stroke: #fff;
+    stroke-width: 3px;
+    stroke-opacity: .8;
+  }
+  .oppia-graph-viz .node rect {
+    fill: #fff;
+    pointer-events: all;
+    stroke: rgb(0, 0, 0);
+    -moz-transition: all 0.15s;
+    -webkit-transition: all 0.15s;
+    -o-transition: all 0.15s;
+    transition: all 0.15s;
+  }
+
+  .oppia-graph-viz .node rect.normal-node:hover {
+    fill: #C8C4C4;
+  }
+  
+  .oppia-graph-viz .legend-bar {
+    fill: url('#nodegradient');
+    stroke: black;
+    stroke-width: 0.5;
+  }
+
+</style>
+
+<div style="height: <[getGraphHeightInPixels()]>px;" ng-show="graphLoaded">
+  <svg class="oppia-graph-viz" width="100%" height="100%">
```

Comment:
If this is meant to be an entirely different component, then use a different "main class", like readonly-directed-graph (or whatever you decide to call it).